client.go

```diff
@@ -135,7 +135,10 @@ func (c *Client[Req, Res]) CallClientStream(ctx context.Context) *ClientStreamFo
 	if c.err != nil {
 		return &ClientStreamForClient[Req, Res]{err: c.err}
 	}
-	return &ClientStreamForClient[Req, Res]{conn: c.newConn(ctx, StreamTypeClient, nil)}
+	return &ClientStreamForClient[Req, Res]{
+		conn:   c.newConn(ctx, StreamTypeClient, nil),
+		config: c.config,
```

Comment:
I've passed around a config object. Should this be limited to `Initializer`?

Reply:
I'd say yes. Config structs are nice on the outer edge of the program, but they have a _lot_ of random stuff on them that shouldn't be hobo-ing its way through every struct.

Reply:
See #640